### PR TITLE
Process Scott PD PPRR 2021

### PR DESCRIPTION
**Pull request by ayyubibrahimi**

### CODE REVIEW

**Comment by ayyubibrahimi** [comment]
@pckhoi can't get the correct regex logic for the middle initial in the split_name function

File: clean/scott_pd_pprr.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+
+def split_name(df):
+    series = df.full_name.fillna("").str.strip()
+    for col, pat in [("first_name", r"^([\w'-]+)(.*)$"), ("middle_initial", r"^(\w\.) (.*)$")]:
+        names = series[series.str.match(pat)].str.extract(pat)
+        df.loc[series.str.match(pat), col] = names[0]
+        series = series.str.replace(pat, r"\2").str.strip()
+    df.loc[:, "last_name"] = series
+    return df.drop(columns=["full_name"])
```
Comment (pckhoi) [inline clean/scott_pd_pprr.py:20]: @ayyubibrahimi try this:
```python
df.full_name.str.strip().str.lower().str.extract(r'^([^ ]+) (?:([^ ]+) )?(.+)')
```

Comment (pckhoi) [reply clean/scott_pd_pprr.py:20]: Regex skill is very valuable. I recommend reading the [Mastering Regex](https://www.amazon.com/Mastering-Regular-Expressions-Understand-Productive-ebook/dp/B007I8S1X0/ref=tmm_kin_swatch_0?_encoding=UTF8&qid=&sr=) book. You will find that overtime the payback is far far greater than this little effort.

Comment (ayyubibrahimi) [reply clean/scott_pd_pprr.py:20]: Just got the book. Thanks!

Comment (ayyubibrahimi) [reply clean/scott_pd_pprr.py:20]: Didn't mean to push to main. Kept getting error messages as I pushed to scott and somehow slipped up and pushed to main accidently. Looking for ways to undo it now.

Comment (ayyubibrahimi) [reply clean/scott_pd_pprr.py:20]: I was able to correct the issue in the "middle_initial" column with: ("middle_initial", r"^(\w\.|\w\s)(.*)$")

**Pull request closed without merging**